src/archivematicaCommon/lib/archivesspace/client.py

```diff
@@ -486,27 +487,46 @@ def add_digital_object(self, parent_archival_object, dashboard_uuid, title="", i
             "notes": [{
                 "jsonmodel_type": "note_digital_object",
                 "type": "originalsloc",
-                "content": [dashboard_uuid]
+                "content": [dashboard_uuid],
+                "publish": True,
             }],
             "restrictions": restricted,
             "subjects": parent_record['subjects'],
             "linked_agents": parent_record['linked_agents'],
         }
 
+        i = 0
+        for pnote in parent_record["notes"]:
+            content = []
+            for subnote in pnote["subnotes"]:
+                content.append(subnote["content"])
+
+            new_object["notes"].append({
+                "jsonmodel_type": "note_digital_object",
+                "type": pnote["type"],
```

Comment:
Like we discussed in person today, not all note types supported by `note_multipart` are supported by `note_digital_object`, so ArchivesSpace is likely to reject some notes copied to records this way. The relevant types are in the `note_digital_object_type` and `note_multipart_type` enums, which are probably fetchable using the API.

You can list all enumerations (and their values) via the `/config/enumerations` API. `note_digital_object_type` is located at `/config/enumerations/44`, and `note_multipart_type` is located at `/config/enumerations/45`.